### ISSUE TITLE
Each editor tab should maintain its own edit mode selection

### ISSUE THREAD

**Issue by x87** [sannybuilder/dev#87]
After #74 each open document could have different active extensions which is difficult to implement if the edit mode configuration is shared.

- [x] edit mode data lives in editor (syntax manager)

- [x] closed files should restore edit mode on open
  - [x] resolve a collision when a MRU file is opened via CLI with edit mode, CLI should control over settings.ini
- [x] new tab inherits currently selected edit mode
  - [x] unclear behavior on opening a compiled script (there is no editor yet) - keep existing behavior, use currently active mode
- [x] hot reload modes.xml - update all tabs
- [x] test $O directive - ini reload
- [x] reload templates in all tabs when a new shared template is added
- [x] show ERR_0001, MSG_0003 only once per session
- [x] refresh models (IDE) when game path is updated for all tabs with the same mode (nice to have, could be a separate feature)
